odoo/addons/base/models/ir_module.py

```diff
@@ -642,6 +642,18 @@ def button_upgrade(self):
         self.update_list()
 
         todo = list(self)
+        if 'base' in self.mapped('name'):
+            # If an installed module is only present in the dependency graph through
+            # a new, uninstalled dependency, it will not have been selected yet.
+            # An update of 'base' should also update these modules, and as a consequence,
+            # install the new dependency.
+            todo += list(self.search(
+                [
+                    ('state', '=', 'installed'),
+                    ('name', '!=', 'studio_customization'),
+                    ('id', 'not in', self.ids),
+                ]
+            ))
```

Comment:
May I suggest a slightly more compact code?
```suggestion
            todo.extend(self.search([
                ('state', '=', 'installed'),
                ('name', '!=', 'studio_customization'),
                ('id', 'not in', self.ids),
            ]))
```

Reply:
Thanks, taken! Do I need to squash commits?